### PR TITLE
Follow changes in the Event Timing API in Chrome 77.

### LINKED ISSUE #23900: Support First Input Delay changes in Chrome 77
The `entryType` for the first input delay PerformanceEntry is changing from `firstInput` to `first-input` in Chrome 77 (see [chrome commit](https://github.com/chromium/chromium/commit/8bae97ee230d843dc27336679257348f0a66a44c#diff-012ea1d48b2f80f230c99f48d06eeca2)). 

We should follow these changes while supporting both names across the new version release.

### PR DESCRIPTION
**Pull request by ericandrewlewis**
Fixes #23900

### CODE REVIEW

File: test/unit/test-performance.js [hunk 1]
```diff
@@ -985,6 +988,41 @@ describes.realWin('PeformanceObserver metrics', {amp: true}, env => {
...
+
+      const perf = Services.performanceFor(env.win);
+
+      expect(perf.events_.length).to.equal(1);
+      expect(perf.events_[0]).to.be.jsonEqual({
+        label: 'fid',
+        delta: 3,
+      });
+
+      delete env.win.PerformanceEventTiming;
```
Comment (npm1) [inline test/unit/test-performance.js]: Not needed?

File: src/service/performance-impl.js [hunk 2]
```diff
@@ -337,14 +363,24 @@ export class Performance {
...
+    if (this.supportsEventTimingAPIv76_) {
       // Programmatically read once as currently PerformanceObserver does not
       // report past entries as of Chrome 61.
       // https://bugs.chromium.org/p/chromium/issues/detail?id=725567
       this.win.performance.getEntriesByType('firstInput').forEach(processEntry);
       entryTypesToObserve.push('firstInput');
     }
 
+    if (this.supportsEventTimingAPIv77_) {
+      // Programmatically read once as currently PerformanceObserver does not
```
[the remark this answers is not in the document]
Comment (ericandrewlewis) [reply src/service/performance-impl.js]: Done in a8b3b8e8b0fe40cd95dea1f441d499dca8b995c8

File: test/unit/test-performance.js (hunk 1 above)
Comment (ericandrewlewis) [reply test/unit/test-performance.js]: Done in a8b3b8e8b0fe40cd95dea1f441d499dca8b995c8

Comment (ericandrewlewis) [reply test/unit/test-performance.js]: Done in a8b3b8e8b0fe40cd95dea1f441d499dca8b995c8

File: src/service/performance-impl.js [hunk 3]
```diff
@@ -372,13 +372,16 @@ export class Performance {
...
 
     if (this.supportsEventTimingAPIv77_) {
-      // Programmatically read once as currently PerformanceObserver does not
-      // report past entries as of Chrome 61.
-      // https://bugs.chromium.org/p/chromium/issues/detail?id=725567
-      this.win.performance
-        .getEntriesByType('first-input')
-        .forEach(processEntry);
-      entryTypesToObserve.push('first-input');
+      // Layout shift entries are not available from the Performance Timeline
```
Comment (npm1) [inline src/service/performance-impl.js]: This comment does not belong here.

**Comment by ericandrewlewis** [comment]
@npm1 said:
> LGTM though in the future we can move to just using PerformanceObserver with the buffered flag set (this is required for largest-contentful-paint, for example).

Sounds good. Is it safe to use the `buffered: true` option in a `PerformanceObserverInit` dictionary across Safari, FF, and older Chrome?

**Comment by npm1** [comment]
> Sounds good. Is it safe to use the `buffered: true` option in a `PerformanceObserverInit` dictionary across Safari, FF, and older Chrome?

Depends what you mean by safe... I believe the buffered flag, or even observing via 'type' instead of 'entryTypes', is not yet supported in other browsers (or older Chrome versions). So using a buffered flag observer can only be used when the buffered flag (and the 'type' option) is supported. But adding things to a dictionary that the browser should not cause problems as far as I know.

File: src/service/performance-impl.js [hunk 4]
```diff
@@ -337,14 +363,25 @@ export class Performance {
...
-    if (this.win.PerformanceEventTiming) {
+    if (this.supportsEventTimingAPIv76_) {
       // Programmatically read once as currently PerformanceObserver does not
       // report past entries as of Chrome 61.
       // https://bugs.chromium.org/p/chromium/issues/detail?id=725567
       this.win.performance.getEntriesByType('firstInput').forEach(processEntry);
       entryTypesToObserve.push('firstInput');
     }
 
+    if (this.supportsEventTimingAPIv77_) {
```
[the remark this answers is not in the document]
Comment (ericandrewlewis) [reply src/service/performance-impl.js]: Yes, and the API implementation is different in each

File: src/service/performance-impl.js [hunk 5]
```diff
@@ -161,6 +161,28 @@ export class Performance {
...
 
+    /**
+     * Whether the user agent supports the Event Timing API that shipped
+     * with Chrome 76.
+     *
+     * @private {boolean}
+     */
+    this.supportsEventTimingAPIv76_ =
+      this.win.PerformanceObserver &&
+      this.win.PerformanceObserver.supportedEntryTypes &&
```
[the remark this answers is not in the document]
Comment (ericandrewlewis) [reply src/service/performance-impl.js]: Done.

**Pull request merged**